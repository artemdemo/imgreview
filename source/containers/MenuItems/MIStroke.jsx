import React from 'react';
import { connect } from 'react-redux';
import MainMenuItem from '../../components/MainMenu/MainMenuItem';
import ColorSelector from '../ColorSelector/ColorSelector.async';
import { showColorPicker } from '../../model/shapes/shapesActions';

class MIStroke extends React.PureComponent {
    onClick = () => {
        const { showColorPicker } = this.props;
        showColorPicker();
    };

    render() {
        const { canvas, shapes } = this.props;
        const item = {
            id: 'color-selector',
            name: 'Color selector',
        };
        return (
            <React.Fragment>
                <MainMenuItem
                    item={item}
                    onClick={this.onClick}
                    disabled={canvas.image == null}
                    key={`main-menu-item__${item.id}`}
                >
                    <div
                        className='main-menu-color'
                        style={{
                            backgroundColor: shapes.stroke,
                        }}
                    />
                </MainMenuItem>
                <ColorSelector />
            </React.Fragment>
        );
    }
}

export default connect(
    state => ({
        canvas: state.canvas,
        shapes: state.shapes,
    }),
    {
        showColorPicker,
    }
)(MIStroke);
