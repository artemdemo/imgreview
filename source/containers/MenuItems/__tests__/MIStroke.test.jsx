import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MIStroke from '../MIStroke';

jest.mock('react-redux');
jest.mock('../../../components/MainMenu/MainMenuItem');
jest.mock('../../ColorSelector/ColorSelector');

describe('MIStroke', () => {
    jest.clearAllMocks();
    const reactReduxMock = require('react-redux');

    it('should render', () => {
        const state = {
            canvas: {
                image: true,
            },
            shapes: {},
        };
        const tree = renderer.create(
            <MIStroke
                canvas={state.canvas}
                shapes={state.shapes}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
        expect(reactReduxMock.__getLastMaps().mapStateToProps(state)).toEqual({
            canvas: state.canvas,
            shapes: state.shapes,
        });
    });

    it('should be disabled', () => {
        const state = {
            canvas: {
                image: null,
            },
            shapes: {},
        };
        const tree = renderer.create(
            <MIStroke
                canvas={state.canvas}
                shapes={state.shapes}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should handle click', () => {
        const showColorPickerMock = jest.fn();
        const state = {
            canvas: {
                image: true,
            },
            shapes: {},
        };
        const wrapper = mount(
            <MIStroke
                canvas={state.canvas}
                shapes={state.shapes}
                showColorPicker={showColorPickerMock}
            />
        );
        wrapper.find('[data-mock="MainMenuItem"]').simulate('click');
        expect(showColorPickerMock).toBeCalled();
    });
});
