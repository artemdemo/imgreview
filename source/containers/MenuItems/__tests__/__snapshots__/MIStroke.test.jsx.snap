// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MIStroke should be disabled 1`] = `
Array [
  <div
    data-mock="MainMenuItem"
    onClick={[Function]}
  >
    <div
      className="main-menu-color"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
  </div>,
  <div
    data-mock="ColorSelector"
  />,
]
`;

exports[`MIStroke should render 1`] = `
Array [
  <div
    data-mock="MainMenuItem"
    onClick={[Function]}
  >
    <div
      className="main-menu-color"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
  </div>,
  <div
    data-mock="ColorSelector"
  />,
]
`;
